### PR TITLE
Issue # 43 - A user should be able to be directed to a list and view a list name 

### LINKED ISSUE #43: Issue # 43. As a user, I want to click on a list and view the name of the list I am viewing when redirected.
**Summary** 

As a user, I need to be able to view the name of the shopping list I am viewing to avoid confusion and ensure I am purchasing from the correct list. 

**Acceptance criteria**

- [ ] The user is immediately redirected to the list after clicking the name of each list. 
- [ ] The name of the list appears at the top of each list view when a user clicks.

### PR DESCRIPTION
**Pull request by eternalmaha**
## Description

Reopening up a new PR again finally closing out Issue #43 after it was opened again and a reverted PR was merged . Fixed bug issues of duplicate home page rendering and list nav bar still appearing. 

In this PR, users should be able to directly navigate to lists after clicking list buttons and list name should appear at the top of list view. This PR also highlighted a new point: list item data will only fetch new lists since our ListItemModel Type was strategically updated in a previously merged PR to include the newly defined parameter: itemQuantity. Incredible team work and insight from the whole team to uncover this new detail! 

## Related Issue

closes #43 

## Acceptance Criteria

- [x] The user is immediately redirected to the list after clicking the name of each list.
- [x] The name of the list appears at the top of each list view when a user clicks.
- [x] The user is still able to view each list from the mange-list view.

## Type of Changes

`enhancement` 

## Updates

https://github.com/user-attachments/assets/d181301d-3699-4bc6-ace7-21859f96a8b6

## Testing Steps / QA Criteria

1. Sign into application
2. Select or create a list 
3. List name will appear the top of list view
4. If list is newly created, list items will appear
5. Else, list items will not appear if it is an old list

### CODE REVIEW

**Comment by eternalmaha** [comment]
Everything appears to be fine so far. Only issue is fetching the data to render in Lists. I realized what the issue was. The list data is passed and fetched properly in Home.tsx and AuthenticatedHome.tsx. The issue was that our fetched items have a different data structure than what our ListItemModel type expects. Our ListItemModel type is expecting the item.quantity property but our fetched items do not have that. When I commented out the itemQuantity defined property in our ListItemModelType, the List component is able to fetch the ListItem data properly…now to work on defining itemQuantity properly in a more appropriate way?

**Comment by eternalmaha** [comment]
Per the conversations in slack with the team, we have come to the conclusion that the issue does not happen with new lists!!! That makes so much sense now that i think about it. Those old lists had already been created with their defined parameters without the item quantity parameter which is why they lists failed to fetch those old list items. I  will go ahead and mark my PR ready for review to then merge my PR. Thank you ever for the perception!

File: src/api/firebase.ts [hunk 1]
```diff
@@ -125,13 +125,15 @@ export function useShoppingListData(listPath: string | null) {
...
 				const item = docSnapshot.data();
+				console.log("Fetched item:", item);
 
 				// The document's id is not in the data,
 				// but it is very useful, so we add it to the data ourselves.
 				item.id = docSnapshot.id;
 
 				const decoded = ListItemModel.decode(item);
 				if (isLeft(decoded)) {
+					console.error("Validation failed for item:", item);
```
Comment (alex-andria) [inline src/api/firebase.ts:136]: Don't forget to remove all console logs!

**Review by alex-andria — APPROVED** [review]
Great work Maha! I see that the previous rendering half-second flash issue is resolved. This is a helpful feature for your team's app!

**Review by bbland1 — APPROVED** [review]

**Pull request merged**